common/pipeline_storage/src/test/scala/uk/ac/wellcome/pipeline_storage/PipelineStorageStreamTest.scala

```diff
@@ -0,0 +1,248 @@
+package uk.ac.wellcome.pipeline_storage
+
+import com.sksamuel.elastic4s.ElasticDsl._
+import com.sksamuel.elastic4s.requests.indexes.admin.IndexExistsResponse
+import com.sksamuel.elastic4s.{ElasticClient, Index, Response}
+import org.scalatest.concurrent.PatienceConfiguration.Timeout
+import org.scalatest.funspec.AnyFunSpec
+import uk.ac.wellcome.akka.fixtures.Akka
+import uk.ac.wellcome.elasticsearch.{ElasticClientBuilder, NoStrictMapping}
+import uk.ac.wellcome.elasticsearch.test.fixtures.ElasticsearchFixtures
+import uk.ac.wellcome.fixtures.{RandomGenerators, TestWith}
+import uk.ac.wellcome.json.JsonUtil
+import uk.ac.wellcome.json.JsonUtil._
+import uk.ac.wellcome.messaging.fixtures.SQS
+import uk.ac.wellcome.messaging.fixtures.SQS.QueuePair
+import uk.ac.wellcome.messaging.memory.MemoryMessageSender
+import uk.ac.wellcome.messaging.sns.NotificationMessage
+import uk.ac.wellcome.models.work.generators.IdentifiersGenerators
+import uk.ac.wellcome.pipeline_storage.fixtures.{
+  ElasticIndexerFixtures,
+  SampleDocument
+}
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class PipelineStorageStreamTest
+    extends AnyFunSpec
+    with ElasticIndexerFixtures
+    with IdentifiersGenerators
+    with RandomGenerators
+    with ElasticsearchFixtures
+    with SQS
+    with Akka {
+
+  val pipelineStorageConfig = PipelineStorageConfig(1, 1 seconds, 10)
+  def indexer(index: Index, elasticClient: ElasticClient = elasticClient) =
+    new ElasticIndexer[SampleDocument](elasticClient, index, NoStrictMapping)
+  def withPipelineStream[R](indexer: Indexer[SampleDocument],
+                            visibilityTimeout: Int = 1,
+                            pipelineStorageConfig: PipelineStorageConfig =
+                              pipelineStorageConfig)(
+    testWith: TestWith[
+      (PipelineStorageStream[NotificationMessage, SampleDocument, String],
+       QueuePair,
+       MemoryMessageSender),
+      R]): R =
+    withActorSystem { implicit ac =>
+      withLocalSqsQueuePair(visibilityTimeout) {
+        case q @ QueuePair(queue, _) =>
```

Comment:
Why are you creating a queue pair if you never use the DLQ?

Reply:
we do, just not here in the `with` method. We pass the whole `QueuePair` in `testWith` so each test can use it